test/e2e/test-controller-image-update.sh

```diff
@@ -0,0 +1,158 @@
+    exit 1
+fi
+echo "Current image: $CURRENT_IMAGE"
+
+# Get current pod names to verify they change
+echo "3. Getting current pod names..."
+CURRENT_PODS=$(kubectl get pods -n kubeflex-system -l control-plane=controller-manager -o jsonpath='{.items[*].metadata.name}')
+echo "Current pods: $CURRENT_PODS"
+
+# Set a unique image tag to force a change
+echo "4. Setting unique image tag for testing..."
+export IMAGE_TAG="e2e-test-$(date +%s)"
+echo "Using image tag: $IMAGE_TAG"
+
+# Run the make install-local-chart command
+echo "5. Running make install-local-chart..."
+make install-local-chart
+
+# Wait for the deployment to update with proper timeout and status checking
+echo "6. Waiting for deployment to update..."
+if ! kubectl rollout status deployment/kubeflex-controller-manager -n kubeflex-system --timeout=180s; then
+    echo "ERROR: Deployment rollout failed or timed out"
+    echo "Deployment status:"
+    kubectl describe deployment kubeflex-controller-manager -n kubeflex-system
+    echo "Pod status:"
+    kubectl get pods -n kubeflex-system -l control-plane=controller-manager
+    exit 1
+fi
+
+# Get the new image
+echo "7. Getting new controller manager image..."
+NEW_IMAGE=$(kubectl get deployment kubeflex-controller-manager -n kubeflex-system -o jsonpath='{.spec.template.spec.containers[?(@.name=="manager")].image}')
+if [ -z "$NEW_IMAGE" ]; then
+    echo "ERROR: Could not get new image from deployment"
+    exit 1
+fi
+echo "New image: $NEW_IMAGE"
+
+# Get new pod names
+echo "8. Getting new pod names..."
+NEW_PODS=$(kubectl get pods -n kubeflex-system -l control-plane=controller-manager -o jsonpath='{.items[*].metadata.name}')
+echo "New pods: $NEW_PODS"
+
+# Wait for deployment rollout to complete and all pods to be ready
+echo "9. Waiting for deployment rollout to complete..."
+if ! kubectl rollout status deployment/kubeflex-controller-manager -n kubeflex-system --timeout=300s; then
+    echo "ERROR: Deployment rollout failed or timed out"
+    echo "Deployment status:"
+    kubectl describe deployment kubeflex-controller-manager -n kubeflex-system
+    exit 1
+fi
+
+# Wait for all pods to be ready
+echo "10. Waiting for all pods to be ready..."
+if ! kubectl wait --for=condition=Ready pods -l control-plane=controller-manager -n kubeflex-system --timeout=120s; then
+    echo "ERROR: Not all pods are ready within timeout"
+    echo "Pod status:"
+    kubectl get pods -n kubeflex-system -l control-plane=controller-manager
+    echo "Pod events:"
+    kubectl describe pods -n kubeflex-system -l control-plane=controller-manager
+    exit 1
+fi
+
+# Verify the fix worked
+echo "11. Verifying the fix..."
+
+# Check if image changed
+if [ "$CURRENT_IMAGE" = "$NEW_IMAGE" ]; then
+    echo " FAILURE: Image did not change"
+    echo "   Current: $CURRENT_IMAGE"
+    echo "   New: $NEW_IMAGE"
+    exit 1
+fi
+
+# Check if pods changed (new pods created)
+if [ "$CURRENT_PODS" = "$NEW_PODS" ]; then
```

Comment:
Comparing pod names as space-separated strings might not reliably detect pod changes if pod ordering changes or there are extra pods. Please do comparison with sorted lists or a more robust set comparison.